### PR TITLE
Added command to deploy generic evm contract via `seid tx evm deploy-contract`

### PR DESCRIPTION
**Pull request by maoueh**
## Describe your changes and provide context

This is similar to `deploy-erc20` but uses a user defined `.abi` and `.bin` file.

## Testing performed to validate your change

I've run the method and it works.

### CODE REVIEW

File: x/evm/client/cli/tx.go [hunk 1]
```diff
@@ -207,6 +209,101 @@ type Response struct {
 	Result  string `json:"result"`
 }
 
+func CmdDeployContract() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deploy-contract <abi> <bin> [<constructor_input_hex>] --from=<sender> --gas-fee-cap=<cap> --gas-limt=<limit> --evm-chain-id=<chain-id> --evm-rpc=<url>",
```
Comment (philipsu522) [inline x/evm/client/cli/tx.go]: i don't think evm-chain-id is a needed flag so we can remove it to avoid confusion

**Review by philipsu522 — APPROVED** [review]

**Review by codchen — APPROVED** [review]

**Comment by maoueh** [comment]
Rebase against `seiv2` now.

**Review by philipsu522 — APPROVED** [review]

**Comment by maoueh** [comment]
How do I get this branch merged?

**Comment by maoueh** [comment]
I'm closing this since `seiv2` was rebased which is causing tons of warnings.

**Pull request closed without merging**